src/MonitoringFunctions/Functions/DryRunUrlChecker.cs

```diff
@@ -0,0 +1,126 @@
+        }
+
+        [FunctionName("DryRun3_1")]
+        public static async Task Run3_1Async([TimerTrigger("0 */30 * * * *")] TimerInfo myTimer, ILogger log)
+        {
+            string monitorName = "dry_run_3_1";
+            string cmdArgs = "-c 3.1";
+
+            await ExecuteDryRunCheckAndReportUrlAccessAsync(log, monitorName, cmdArgs).ConfigureAwait(false);
+        }
+        
+        [FunctionName("DryRun3_0Runtime")]
+        public static async Task Run3_0RuntimeAsync([TimerTrigger("0 */30 * * * *")] TimerInfo myTimer, ILogger log)
+        {
+            string monitorName = "dry_run_3_0_runtime";
+            string cmdArgs = "-c 3.0 -Runtime dotnet";
+
+            await ExecuteDryRunCheckAndReportUrlAccessAsync(log, monitorName, cmdArgs).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Executes the Ps1 script with DryDun switch,
+        /// Parses the output to acquire primary and legacy Urls,
+        /// Tests the primary Url to see if it is available,
+        /// Reports the results to the data service provided.
+        /// </summary>
+        internal static async Task ExecuteDryRunCheckAndReportUrlAccessAsync(ILogger log, string monitorName, string additionalCmdArgs,
+            CancellationToken cancellationToken = default)
+        {
+            string scriptName = "dotnet-install.ps1";
+            string commandLineArgs = $"-DryRun {additionalCmdArgs}";
+
+            using IDataService dataService = new DataServiceFactory().GetDataService();
+            
+            // Execute the script;
+            ScriptExecutionResult results = await HelperMethods.ExecuteInstallScriptPs1Async(commandLineArgs).ConfigureAwait(false);
+
+            log.LogInformation($"Ouput stream: {results.Output}");
+
+            if (!string.IsNullOrWhiteSpace(results.Error))
+            {
+                log.LogError($"Error stream: {results.Error}");
+                await dataService.ReportScriptExecutionAsync(monitorName, scriptName, commandLineArgs, results.Error, cancellationToken)
+                    .ConfigureAwait(false);
+                return;
+            }
+
+            // Parse the output
+            ScriptDryRunResult dryRunResults = ParseDryRunOutput(results.Output);
+
+            if (string.IsNullOrWhiteSpace(dryRunResults.PrimaryUrl))
+            {
+                log.LogError($"Primary Url was not found for channel {additionalCmdArgs}");
+                await dataService.ReportScriptExecutionAsync(monitorName, scriptName, commandLineArgs,
+                    "Failed to parse primary url from the following DryRun execution output: " + results.Output
+                    , cancellationToken).ConfigureAwait(false);
+                return;
+            }
+
+            // Validate URL accessibility
+            await HelperMethods.CheckAndReportUrlAccessAsync(log, monitorName, dryRunResults.PrimaryUrl, dataService);
```

Comment:
what is the "check" signal?  Does the method throw if the check fails?

Reply:
Yes, exceptions report the failures. Updated the documentation to make it more clear.